internal/controller/pkg/revision/establisher.go

```diff
@@ -73,6 +80,27 @@ type currentDesired struct {
 func (e *APIEstablisher) Establish(ctx context.Context, objs []runtime.Object, parent resource.Object, control bool) ([]xpv1.TypedReference, error) { // nolint:gocyclo
 	allObjs := []currentDesired{}
 	resourceRefs := []xpv1.TypedReference{}
+	var rev v1.PackageRevision
```

Comment:
@hasheddan Is it fine if we change `parent resource.Object` in the interface signature to `v1.PackageRevision` since it's guaranteed to be a revision in all cases?

Reply:
That is fine by me, but do we gain anything from making it a more scoped interface? Do you feel like it is clearer?

Reply:
We need to get the webhook tls secret name, so if it's `resource.Object` then we'd have to cast it to `PackageRevision` interface, which we know always is the case. So, it's more like reducing its generality to where it actually is. Also, it's a bit easier to understand the code when you know the parent is a revision and not package.

Reply:
@muvaf sounds good!